idemp-base-test/src/org/bandahealth/idempiere/base/test/modelevent/OrderModelEventTest.java

```diff
@@ -0,0 +1,55 @@
+package org.bandahealth.idempiere.base.test.modelevent;
+
+import com.chuboe.test.populate.ChuBoeCreateEntity;
+import com.chuboe.test.populate.ChuBoePopulateFactoryVO;
+import com.chuboe.test.populate.ChuBoePopulateVO;
+import com.chuboe.test.populate.IPopulateAnnotation;
+import org.bandahealth.idempiere.base.model.MDocType_BH;
+import org.compiere.model.MOrder;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OrderModelEventTest extends ChuBoePopulateFactoryVO {
+	@IPopulateAnnotation.CanRunBeforeClass
+	public void prepareIt() throws Exception {
+		ChuBoePopulateVO valueObject = new ChuBoePopulateVO();
+		valueObject.prepareIt(getScenarioName(), true, get_TrxName());
+		assertThat("VO validation gives no errors", valueObject.getErrorMessage(), is(nullValue()));
+
+		valueObject.setStepName("Open needed periods");
+		ChuBoeCreateEntity.createAndOpenAllFiscalYears(valueObject);
+		commitEx();
+	}
+
+	@IPopulateAnnotation.CanRun
+	public void orderIsCreatedCorrectlyWithModelEvents() throws Exception {
+		ChuBoePopulateVO valueObject = new ChuBoePopulateVO();
```

Comment:
I'd say we don't have to open periods if a test doesn't expect to be completing any documents. Otherwise, the periods need to be open (and we have to repeat it in every class in case someone installs a new DB, then just decides to run that one test - happened to me...).